File: mssp/examples/011-store-boundary/src/SCL/policy.mjs
// Which medium and handout this deployment uses, and what it refuses to ship.
import fs from "node:fs";
import path from "node:path";
import { fileURLToPath } from "node:url";

const here = path.dirname(fileURLToPath(import.meta.url));
const config = JSON.parse(fs.readFileSync(path.join(here, "policy.json"), "utf8"));

export const medium = () => config.medium;
export const handout = () => config.handout;
export const liveReferenceIsFatal = () => Boolean(config.live_reference_is_fatal);

File: mssp/archaeology/011-cpython-shelve/src/SCL/policy.py
"""Which handout this deployment uses, and what it refuses to let pass."""
import json
import pathlib

_C = json.loads((pathlib.Path(__file__).parent / "policy.json").read_text(encoding="utf-8"))

handout = lambda: _C["handout"]                                                  # noqa: E731
declaration_must_match_behaviour = lambda: bool(_C["declaration_must_match_behaviour"])  # noqa: E731
recut_must_match_upstream = lambda: bool(_C["recut_must_match_upstream"])        # noqa: E731
